src/main/ApplicationImpl.cpp

```diff
@@ -198,6 +206,10 @@ ApplicationImpl::getNetworkID() const
 ApplicationImpl::~ApplicationImpl()
 {
     LOG(INFO) << "Application destructing";
+    if (mNtpSynchronizationChecker)
+    {
+        mNtpSynchronizationChecker->shutdown();
```

Comment:
I was just following places where mProcessManager->shutdown() is called, as
mNtpSynchronizationChecker is similar to mProcessManager. Checking if this is really needed would require separate investigation.